### ISSUE TITLE
Allow to disable install completions altogether

### ISSUE THREAD

**Issue by martin-braun** [oven-sh/bun#11179]
### What is the problem this feature would solve?

Checking on the installer script, I see no way to skip the install completions, other than installing through an unsupported shell. I carefully craft and maintain my dot files, I wish to do the completions by hand. Also #10897 gives me a bitter aftertaste.

### What is the feature you are proposing to solve the problem?

A way to opt-out of install completions altogether, on installation or any update. For this I would propose that we can set `export BUN_SKIP_INSTALL_COMPLETIONS=1` in our dot files. I don't wish `bun` to tinker with my configuration.

### What alternatives have you considered?

I removed the necessary parts from the installer script prior install, but seeing that an update might cause the completions to run anyways makes me ask for a better solution. 

Bun is for programmers. You can be a toy for Windows users, but not for Unix users, please. I would even support to get rid of install completions altogether at this point.

**Comment by YogiLiu** [comment]
I think bun can remind users to write commands into the corresponding configuration files instead of rudely making decisions for users, not just only completions, but also PATH.

**Comment by YogiLiu** [comment]
Many Unix users have a dedicated directory structure to manage shell configurations, and currently such behavior can really cause trouble.

**Comment by sirenkovladd** [comment]
Like workaround, you can add text "# bun completions" in your .zshrc or .bashrc for avoid auto install completions

https://github.com/oven-sh/bun/blob/bun-v1.1.12/src/cli/install_completions_command.zig#L540

**Comment by YogiLiu** [comment]
@sakibhasancse  I think this is just a temporary solution. At least for users of [oh-my-zsh](https://ohmyz.sh/), the solution using environment variables will be better than it.

**Comment by martin-braun** [comment]
@sirenkovladd That probably works, but you see where this is going? Magic, magic, everywhere. No, we should not admire complexity and instead we should strive for simplicity. A tool can instruct me on editing my configuration to make itself work, but it should not try to cover everyone and any setup any dependency upstream. 

Dependency upstream, because of the dependency chain, which is just in the wrong order at this point. First there was the shell configuration, then comes the tool that gets installed into the system. This is a many to one, there are many different shell configurations, but there is only one `bun`. `bun` has to stay in its limits, because it's not its job to tinker with dependencies it knows nothing about. It really tries to guess and cover anything by brute force, the very opposite of simplicity.

I might have a file with completions and my comments don't include `completions`, because the file's name implies that. I will not disrupt a configuration for something that relies on my configuration. Imagine every tool you install would do this, your rc files would be full with "do not edit this comment" comments all galore.

**Comment by sirenkovladd** [comment]
I agree with you, it's not a solution, like I said, it's just a `workaround`

**Comment by robobun** [comment]
This is the same request tracked in #1272: Bun's completions installer (run by the install script and again by `bun upgrade`) edits the user's shell dotfiles to install completions, and both issues ask for a flag/environment variable to opt out so dotfiles are left alone. A single opt-out for the completions installer resolves both, so it will be tracked in #1272.

Duplicate of #1272

Please add a 👍 or any extra details over there so it all stays in one place. If this is actually a different problem, say so here and we'll reopen.
